fujiji-server/src/controllers/comment.js

```diff
@@ -62,4 +70,56 @@ async function getListingComments(req, res, next) {
   }
 }
 
-module.exports = { postComment, getListingComments };
+// PURPOSE: implement the put comment endpoint
+async function editComment(req, res, next) {
+  const { comment, isHighlighted } = req.body;
+  const { comment_id: commentID } = req.params;
+  const { id } = req.locals.userData;
+  const userID = parseInt(id, 10);
+  const intCommentId = parseInt(commentID, 10);
+
+  const existingComment = await getCommentById(intCommentId);
+
+  if (!existingComment) {
+    return next(
+      new CommentNotFoundError(`comment with id:${commentID} is not found`),
+    );
+  }
+
+  const listingID = existingComment.listing_id;
+  const listing = await getListingById(parseInt(listingID, 10));
+
+  if (!listing) {
+    return next(
+      new ListingNotFound(`listing with id:${listingID} is not found`),
+    );
+  }
+
+  const isCommenter = userID === parseInt(existingComment.user_id, 10) ? 1 : 0;
+
+  const isAuthor = userID === parseInt(listing.user_id, 10) ? 1 : 0;
+
+  try {
+    if (isCommenter) {
+      if (!comment) {
+        return next(new EmptyCommentError());
+      }
+
+      if (comment != existingComment.comment) {
+        await updateCommentById(intCommentId, comment);
+      }
+    }
+    if (isAuthor) {
+      if (isHighlighted != existingComment.isHighlighted) {
+        await highlightsCommentById(intCommentId, isHighlighted);
+      }
+    }
+
+    const updatedComment = await getCommentById(intCommentId);
+    return res.status(200).json({ updatedComment });
+  } catch (err) {
+    return next(new APIError(err, 500));
+  }
+}
```

Comment:
To check if I get this right. If the sellers want to highlight their own comments, they need to include `comment` field in the payload, right? Otherwise, we will return `EmptyCommentError`. 

But, if the sellers highlight other's comment, we don't need to include `comment` in the payload

Reply:
Otherwise, everything lgtm!

Reply:
Yup, you got it correct! The `comment` in the payload can be empty if it's just the seller highlighting others' comment